.github/workflows/main.yaml

```diff
@@ -41,10 +51,102 @@ jobs:
           FLUTTER_DART_HOME: ${{ env.FLUTTER_HOME }}/bin/cache/dart-sdk/bin
         run: |
           export PATH=$PATH:$FLUTTER_DART_HOME
-          ./scripts/ci.sh packages/${{ matrix.package }}
+          ./scripts/ci.sh ${{ matrix.folder }}
 
       - name: Upload coverage to Codecov
         uses: codecov/codecov-action@v1
         with:
           token: ${{ secrets.CODECOV_TOKEN }}
           file: ${{ matrix.package }}.lcov
+
+  # Analyze, test and build an APK for each flutter example.
+  examples-apk:
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      matrix:
+        folder: ['examples/flutter_bloc_with_stream', 'examples/flutter_complex_list', 'examples/flutter_dynamic_form', 'examples/flutter_firebase_login', 'examples/flutter_firestore_todos', 'examples/flutter_form_validation', 'examples/flutter_infinite_list', 'examples/flutter_login', 'examples/flutter_shopping_cart', 'examples/flutter_timer', 'examples/flutter_todos', 'examples/flutter_weather', 'examples/github_search/flutter_github_search', 'packages/flutter_bloc/example', 'performance_tests/flutter_bloc/bloc_builder', 'performance_tests/flutter_bloc/bloc_provider']
+        os: [ubuntu-latest]
+        include:
+          - os: ubuntu-latest
+            flutter_path: /opt/hostedtoolcache/flutter
+      fail-fast: false
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Cache Flutter dependencies
+        uses: actions/cache@v1
+        with:
+          path: ${{ matrix.flutter_path }}
+          key: ${{ runner.OS }}-flutter-cache-${{ env.flutter_version }}
+          restore-keys: |
+            ${{ runner.OS }}-flutter-cache-
+
+      - uses: subosito/flutter-action@v1
+        with:
+          channel: ${{ env.flutter_channel }}
+          flutter-version: ${{ env.flutter_version }}
+
+      - name: Analyze and test
+        env:
+          FLUTTER_DART_HOME: ${{ env.FLUTTER_HOME }}/bin/cache/dart-sdk/bin
+        run: |
+          export PATH=$PATH:$FLUTTER_DART_HOME
+          ./scripts/ci.sh ${{ matrix.folder }}
+
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v1
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }}
+          file: ${{ matrix.package }}.lcov
+
+      - name: Build an APK
+        env:
+          FLUTTER_DART_HOME: ${{ env.FLUTTER_HOME }}/bin/cache/dart-sdk/bin
+        run: |
+          export PATH=$PATH:$FLUTTER_DART_HOME
+          ./scripts/ci_build_apk.sh ${{ matrix.folder }}
+
+  # Analyze, test and build for each web example.
+  examples-web:
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      matrix:
+        folder: ['packages/bloc/example', 'packages/bloc_test/example', 'examples/angular_counter', 'examples/github_search/common_github_search', 'examples/github_search/angular_github_search']
```

Comment:
why is the `bloc/example` and `bloc_test/example` included in `examples-web`? They are pure dart examples.

Reply:
That's true that we don't do anything more than on `packages/*` here, we just do a pub get and then analyze.
Removing these folders.

Reply:
I am not sure removing these folders is a good way. More like creating examples-pure that contains only analysis? It seems to me like you have deleted it.